#  Data preprocessing (screening out variables and removing missing values)

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.pyplot as plt


plt.rcParams['font.sans-serif'] = ['SimHei']  

plt.rcParams['axes.unicode_minus'] = False

In [2]:
df = pd.read_csv('LLCP2017.csv')


In [3]:

selected_columns = ['SEX', '_AGE_G','_RACEGR3','_EDUCAG', '_INCOMG', '_SMOKER3', '_BMI5CAT','_FRTLT1A', '_VEGLT1A', '_TOTINDA', '_RFBING5', '_ECIGSTS','_PHYS14D','_MENT14D','CHECKUP1','_LTASTH1']

new_df = df[selected_columns]

new_df.to_csv('Variable_0.csv', index=False)


print(new_df)

        SEX  _AGE_G  _RACEGR3  _EDUCAG  _INCOMG  _SMOKER3  _BMI5CAT  _FRTLT1A  \
0       2.0     6.0       1.0      4.0      4.0       4.0       3.0       1.0   
1       1.0     6.0       1.0      4.0      5.0       4.0       3.0       1.0   
2       1.0     6.0       1.0      1.0      9.0       4.0       3.0       2.0   
3       2.0     6.0       1.0      1.0      1.0       4.0       3.0       2.0   
4       2.0     6.0       1.0      1.0      1.0       3.0       2.0       1.0   
...     ...     ...       ...      ...      ...       ...       ...       ...   
450011  2.0     2.0       5.0      3.0      9.0       4.0       3.0       1.0   
450012  1.0     6.0       5.0      2.0      1.0       4.0       2.0       2.0   
450013  1.0     4.0       5.0      4.0      5.0       4.0       4.0       2.0   
450014  2.0     2.0       5.0      3.0      1.0       3.0       4.0       2.0   
450015  2.0     2.0       5.0      4.0      4.0       4.0       3.0       2.0   

        _VEGLT1A  _TOTINDA 

In [4]:
df = pd.read_csv('Variable_0.csv', na_values=['', 'NA', 'null', '9','77','99','88'])

In [5]:
print(df['SEX'].isna().sum())
print(df['_RFBING5'].isna().sum())
print(df.info)

284
29611
<bound method DataFrame.info of         SEX  _AGE_G  _RACEGR3  _EDUCAG  _INCOMG  _SMOKER3  _BMI5CAT  _FRTLT1A  \
0       2.0     6.0       1.0      4.0      4.0       4.0       3.0       1.0   
1       1.0     6.0       1.0      4.0      5.0       4.0       3.0       1.0   
2       1.0     6.0       1.0      1.0      NaN       4.0       3.0       2.0   
3       2.0     6.0       1.0      1.0      1.0       4.0       3.0       2.0   
4       2.0     6.0       1.0      1.0      1.0       3.0       2.0       1.0   
...     ...     ...       ...      ...      ...       ...       ...       ...   
450011  2.0     2.0       5.0      3.0      NaN       4.0       3.0       1.0   
450012  1.0     6.0       5.0      2.0      1.0       4.0       2.0       2.0   
450013  1.0     4.0       5.0      4.0      5.0       4.0       4.0       2.0   
450014  2.0     2.0       5.0      3.0      1.0       3.0       4.0       2.0   
450015  2.0     2.0       5.0      4.0      4.0       4.0       3.0

In [6]:
df.columns

Index(['SEX', '_AGE_G', '_RACEGR3', '_EDUCAG', '_INCOMG', '_SMOKER3',
       '_BMI5CAT', '_FRTLT1A', '_VEGLT1A', '_TOTINDA', '_RFBING5', '_ECIGSTS',
       '_PHYS14D', '_MENT14D', 'CHECKUP1', '_LTASTH1'],
      dtype='object')

In [7]:
len(df)

print(df.isnull().sum())

SEX           284
_AGE_G          0
_RACEGR3     8755
_EDUCAG      1707
_INCOMG     75085
_SMOKER3    18724
_BMI5CAT    36446
_FRTLT1A    39357
_VEGLT1A    47460
_TOTINDA    34203
_RFBING5    29611
_ECIGSTS    20646
_PHYS14D     9331
_MENT14D     7203
CHECKUP1      526
_LTASTH1     1463
dtype: int64


In [8]:
df_clean = df.dropna(subset=[ '_SMOKER3','_BMI5CAT','_RFBING5'])  
print("The number of rows after removing missing values:", df_clean.shape[0])
df_clean = df.dropna()
print("The final number of rows of data:", len(df_clean))


The number of rows after removing missing values: 392554
The final number of rows of data: 307208


In [9]:

count_1 = df_clean[df_clean['_LTASTH1'] == 2].shape[0]
print("In df_clean, the number of rows with ASTHMA3 equal to 1 is:", count_1)


In df_clean, the number of rows with ASTHMA3 equal to 1 is: 42549


The proportions have remained basically the same, the original 450,000, 60,000 asthma. The new data set is 307208, 42549 asthma

## Reclassify variables(Mapping)

In [10]:
import pandas as pd

# Create a mapping dictionary to map the original _AGE_G value to 1, 2, 3
age_mapping = {
    1: 1,  # Age 18-24 -> 19-44 (1)
    2: 1,  # Age 25-34 -> 19-44 (1)
    3: 1,  # Age 35-44 -> 19-44 (1)
    4: 2,  # Age 45-54 -> 45-64 (2)
    5: 2,  # Age 55-64 -> 45-64 (2)
    6: 3   # Age 65+ -> 65+ (3)
}

df_clean["_AGE_GROUP"] = df_clean["_AGE_G"].map(age_mapping)



# Create a mapping dictionary to map the original _EDUCAG value to 1, 2
edu_mapping = {
    1: 1,  # Low Education
    2: 2,  # high Education
    3: 2,  # high Education
    4: 2  # high Education
}


df_clean["_EDUCA_GROUP"] = df_clean["_EDUCAG"].map(edu_mapping)



# Create a mapping dictionary to map the original _INCOMG value to 1, 2, 3
income_mapping = {
    1: 1,  # low income
    2: 2,  # mid income
    3: 2,  # mid income
    4: 3,  # high income
    5: 3  # high income
}


df_clean["_INCOM_GROUP"] = df_clean["_INCOMG"].map(income_mapping)



# Create a mapping dictionary to map the original _SMOKER3 value to 1, 2, 3
smoker_mapping = {
    1: 1,  # Current smoker
    2: 1,  # Current smoker
    3: 2,  # Former smoker
    4: 3  # Never smoked
}


df_clean["_SMOKER3_GROUP"] = df_clean["_SMOKER3"].map(smoker_mapping)



# Create a mapping dictionary to map the original _ECIGSTS value to 1, 2, 3
esmoker_mapping = {
    1: 1,  # Current Esmoker
    2: 1,  # Current Esmoker
    3: 2,  # Former Emoker
    4: 3  # Never Esmoked
}


df_clean["_ECIGSTS_GROUP"] = df_clean["_ECIGSTS"].map(esmoker_mapping)



# Create a mapping dictionary to map the original CHECKUP1 value to 1, 2
checkup_mapping = {
    1: 1,  # within 1 year
    2: 2,  # other
    3: 2,  # other
    4: 2,  # other
    7: 2,  # other
    8: 2  # other 
    }


df_clean["CHECKUP1_GROUP"] = df_clean["CHECKUP1"].map(checkup_mapping)

/var/folders/49/vkc7dh1j2mq0ncj_gx61lskh0000gn/T/ipykernel_3712/725741163.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_clean["_AGE_GROUP"] = df_clean["_AGE_G"].map(age_mapping)
/var/folders/49/vkc7dh1j2mq0ncj_gx61lskh0000gn/T/ipykernel_3712/725741163.py:26: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_clean["_EDUCA_GROUP"] = df_clean["_EDUCAG"].map(edu_mapping)
/var/folders/49/vkc7dh1j2mq0ncj_gx61lskh0000gn/T/ipykernel_3712/725741163.py:40: SettingWithCopyWarning: 
A value is trying to be set

In [11]:
# Save the results and create a new CSV
df_clean.to_csv("Variable_1.csv", index=False)

In [12]:
df = pd.read_csv('Variable_1.csv')

print(df.info)

<bound method DataFrame.info of         SEX  _AGE_G  _RACEGR3  _EDUCAG  _INCOMG  _SMOKER3  _BMI5CAT  _FRTLT1A  \
0       2.0     6.0       1.0      4.0      4.0       4.0       3.0       1.0   
1       1.0     6.0       1.0      4.0      5.0       4.0       3.0       1.0   
2       2.0     6.0       1.0      1.0      1.0       3.0       2.0       1.0   
3       1.0     6.0       1.0      2.0      2.0       1.0       3.0       2.0   
4       1.0     6.0       1.0      3.0      5.0       4.0       3.0       1.0   
...     ...     ...       ...      ...      ...       ...       ...       ...   
307203  1.0     2.0       5.0      3.0      3.0       1.0       3.0       1.0   
307204  1.0     6.0       5.0      2.0      1.0       4.0       2.0       2.0   
307205  1.0     4.0       5.0      4.0      5.0       4.0       4.0       2.0   
307206  2.0     2.0       5.0      3.0      1.0       3.0       4.0       2.0   
307207  2.0     2.0       5.0      4.0      4.0       4.0       3.0       2.0

In [15]:
selected_columns_1 = ['SEX', '_AGE_GROUP','_RACEGR3','_EDUCA_GROUP', '_INCOM_GROUP', '_SMOKER3_GROUP', '_BMI5CAT','_FRTLT1A', '_VEGLT1A', '_TOTINDA', '_RFBING5', '_ECIGSTS_GROUP','_PHYS14D','_MENT14D','CHECKUP1_GROUP','_LTASTH1']

new_df_1 = df[selected_columns_1]

new_df_1.to_csv('Variable_final.csv', index=False)
print(new_df_1)

        SEX  _AGE_GROUP  _RACEGR3  _EDUCA_GROUP  _INCOM_GROUP  _SMOKER3_GROUP  \
0       2.0           3       1.0             2             3               3   
1       1.0           3       1.0             2             3               3   
2       2.0           3       1.0             1             1               2   
3       1.0           3       1.0             2             2               1   
4       1.0           3       1.0             2             3               3   
...     ...         ...       ...           ...           ...             ...   
307203  1.0           1       5.0             2             2               1   
307204  1.0           3       5.0             2             1               3   
307205  1.0           2       5.0             2             3               3   
307206  2.0           1       5.0             2             1               2   
307207  2.0           1       5.0             2             3               3   

        _BMI5CAT  _FRTLT1A 

In [ ]:
new_df_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 307208 entries, 0 to 307207
Data columns (total 16 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   SEX             307208 non-null  float64
 1   _AGE_GROUP      307208 non-null  int64  
 2   _RACEGR3        307208 non-null  float64
 3   _EDUCA_GROUP    307208 non-null  int64  
 4   _INCOM_GROUP    307208 non-null  int64  
 5   _SMOKER3_GROUP  307208 non-null  int64  
 6   _BMI5CAT        307208 non-null  float64
 7   _FRTLT1A        307208 non-null  float64
 8   _VEGLT1A        307208 non-null  float64
 9   _TOTINDA        307208 non-null  float64
 10  _RFBING5        307208 non-null  float64
 11  _ECIGSTS_GROUP  307208 non-null  int64  
 12  _PHYS14D        307208 non-null  float64
 13  _MENT14D        307208 non-null  float64
 14  CHECKUP1_GROUP  307208 non-null  int64  
 15  _LTASTH1        307208 non-null  float64
dtypes: float64(10), int64(6)
memory usage: 37.5 MB
